### PR TITLE
Several small fixes to support conda-forge

### PR DESCRIPTION
**Pull request by xylar**
When I tried to build the `mpas_tools` package on conda-forge, I ran into 3 issues:
* compilers (g++ for linux and clang for OSX) don't work well with our simple make file
* in clang, there is a conflict between `std::merge` and the local definition of `merge` in the cell culler (and probably the same issue in g++ but not a fatal error).
* The test for the python interface to the mesh conversion tools needs to use the 'Agg' backend for matplotlib because CI doesn't have a display.

This merge adds support for a cmake build of `mesh_conversion_tools` and fixes the other two issues above in the hopes of making another minor release that will work without patching on conda-forge.

### CODE REVIEW

**Comment by xylar** [comment]
@mark-petersen, if you can verify that the `mesh_conversion_tools` build for you in the "old-fashioned way", that is, with just `make`, I think that would be a sufficiently rigorous test of this.

If you want to test with cmake, the instructions would be:
``` bash
cd mesh_tools/mesh_conversion_tools
mkdir build
cd build
cmake -DCMAKE_BUILD_TYPE=Release ..
￼make
```
This worked for me (once I installed cmake).

Don't worry about the `test_conversion.py`, I verified that this fixed the problem I was having on conda-forge and I don't think you need to test that.

File: mesh_tools/mesh_conversion_tools/mpas_cell_culler.cpp [hunk 1]
```diff
@@ -16,7 +16,7 @@
 
 using namespace std;
 
-enum { merge, invert, preserve };
+enum { mergeOp, invertOp, preserveOp };
```
Comment (xylar) [inline mesh_tools/mesh_conversion_tools/mpas_cell_culler.cpp]: `merge` was conflicting with `std::merge`.  (It is a bad practice to do `using namespace std;` as above, but that's too big a problem for me to want to fix it.)  For consistency, I renamed all 3 operations.

File: mesh_tools/mesh_conversion_tools/mpas_cell_culler.cpp [hunk 2]
```diff
@@ -614,21 +614,21 @@ int outputGridDimensions( const string outputFilename ){/*{{{*/
 	int nCellsNew, nEdgesNew, nVerticesNew;
 	// Return this code to the OS in case of failure.
 	static const int NC_ERR = 2;
-
```
Comment (xylar) [inline mesh_tools/mesh_conversion_tools/mpas_cell_culler.cpp]: This kind of whitespace clean-up happened automatically in my editor.  I can revert it but I think it's not a bad idea to remove it.

**Comment by xylar** [comment]
@mark-petersen, if you could review this quickly today, that would be very helpful.  I really need to get `mpas_tools` onto conda-forge before I can rebuild the `compass_py3.7` environment.

**Comment by mark-petersen** [comment]
<details>
<summary> Tested with intel and gnu on grizzly, and get the same error on both.</summary> <p>

```

gr-fe3:build$ module list

Currently Loaded Modules:
  1) git/2.19.1                    3) intel/17.0.1     5) netcdf/4.4.1            7) pio/1.7.2
  2) python/anaconda-2.7-climate   4) openmpi/1.10.5   6) parallel-netcdf/1.5.0   8) cmake/3.14.0



gr-fe3:build$ cmake -DCMAKE_BUILD_TYPE=Release ..
-- Configuring done
-- Generating done
-- Build files have been written to: /usr/projects/climate/mpeterse/repos/tools/pr_test/mesh_tools/mesh_conversion_tools/build
gr-fe3:build$ make
[  6%] Building CXX object CMakeFiles/MpasMaskCreator.x.dir/mpas_mask_creator.cpp.o
In file included from /turquoise/usr/projects/hpcsoft/toss3/common/x86_64/gcc/5.3.0/include/c++/5.3.0/tuple:35:0,
                 from /usr/projects/climate/mpeterse/repos/tools/pr_test/mesh_tools/mesh_conversion_tools/mpas_mask_creator.cpp:15:
/turquoise/usr/projects/hpcsoft/toss3/common/x86_64/gcc/5.3.0/include/c++/5.3.0/bits/c++0x_warning.h:32:2: error: #error This file requires compiler and library support for the ISO C++ 2011 standard. This support is currently experimental, and must be enabled with the -std=c++11 or -std=gnu++11 compiler options.
 #error This file requires compiler and library support for the \
  ^
/usr/projects/climate/mpeterse/repos/tools/pr_test/mesh_tools/mesh_conversion_tools/mpas_mask_creator.cpp:64:17: error: ‘tuple’ was not declared in this scope
 vector< vector< tuple<string, string, int>> > regionProperties;
                 ^
/usr/projects/climate/mpeterse/repos/tools/pr_test/mesh_tools/mesh_conversion_tools/mpas_mask_creator.cpp:64:42: error: ‘>>’ should be ‘> >’ within a nested template argument list
 vector< vector< tuple<string, string, int>> > regionProperties;
                                          ^
/usr/projects/climate/mpeterse/repos/tools/pr_test/mesh_tools/mesh_conversion_tools/mpas_mask_creator.cpp:64:39: error: wrong number of template arguments (3, should be at least 1)
 vector< vector< tuple<string, string, int>> > regionProperties;
                                       ^
In file included from /turquoise/usr/projects/hpcsoft/toss3/common/x86_64/gcc/5.3.0/include/c++/5.3.0/vector:64:0,
                 from /usr/projects/climate/mpeterse/repos/tools/pr_test/mesh_tools/mesh_conversion_tools/mpas_mask_creator.cpp:6:
/turquoise/usr/projects/hpcsoft/toss3/common/x86_64/gcc/5.3.0/include/c++/5.3.0/bits/stl_vector.h:214:11: note: provided for ‘template<class _Tp, class _Alloc> class std::vector’
     class vector : protected _Vector_base<_Tp, _Alloc>
           ^
/usr/projects/climate/mpeterse/repos/tools/pr_test/mesh_tools/mesh_conversion_tools/mpas_mask_creator.cpp:64:42: error: template argument 1 is invalid
 vector< vector< tuple<string, string, int>> > regionProperties;
                                          ^
/usr/projects/climate/mpeterse/repos/tools/pr_test/mesh_tools/mesh_conversion_tools/mpas_mask_creator.cpp:64:42: error: template argument 2 is invalid
/usr/projects/climate/mpeterse/repos/tools/pr_test/mesh_tools/mesh_conversion_tools/mpas_mask_creator.cpp:64:45: error: expected unqualified-id before ‘>’ token
 vector< vector< tuple<string, string, int>> > regionProperties;
                                             ^
/usr/projects/climate/mpeterse/repos/tools/pr_test/mesh_tools/mesh_conversion_tools/mpas_mask_creator.cpp:73:17: error: ‘tuple’ was not declared in this scope
 vector< vector< tuple<string, string, int>> > transectProperties;
                 ^
/usr/projects/climate/mpeterse/repos/tools/pr_test/mesh_tools/mesh_conversion_tools/mpas_mask_creator.cpp:73:42: error: ‘>>’ should be ‘> >’ within a nested template argument list
 vector< vector< tuple<string, string, int>> > transectProperties;
                                          ^
/usr/projects/climate/mpeterse/repos/tools/pr_test/mesh_tools/mesh_conversion_tools/mpas_mask_creator.cpp:73:39: error: wrong number of template arguments (3, should be at least 1)
 vector< vector< tuple<string, string, int>> > transectProperties;
                                       ^
In file included from /turquoise/usr/projects/hpcsoft/toss3/common/x86_64/gcc/5.3.0/include/c++/5.3.0/vector:64:0,
                 from /usr/projects/climate/mpeterse/repos/tools/pr_test/mesh_tools/mesh_conversion_tools/mpas_mask_creator.cpp:6:
/turquoise/usr/projects/hpcsoft/toss3/common/x86_64/gcc/5.3.0/include/c++/5.3.0/bits/stl_vector.h:214:11: note: provided for ‘template<class _Tp, class _Alloc> class std::vector’
     class vector : protected _Vector_base<_Tp, _Alloc>
           ^
/usr/projects/climate/mpeterse/repos/tools/pr_test/mesh_tools/mesh_conversion_tools/mpas_mask_creator.cpp:73:42: error: template argument 1 is invalid
 vector< vector< tuple<string, string, int>> > transectProperties;
                                          ^
/usr/projects/climate/mpeterse/repos/tools/pr_test/mesh_tools/mesh_conversion_tools/mpas_mask_creator.cpp:73:42: error: template argument 2 is invalid
/usr/projects/climate/mpeterse/repos/tools/pr_test/mesh_tools/mesh_conversion_tools/mpas_mask_creator.cpp:73:45: error: expected unqualified-id before ‘>’ token
 vector< vector< tuple<string, string, int>> > transectProperties;
                                             ^
/usr/projects/climate/mpeterse/repos/tools/pr_test/mesh_tools/mesh_conversion_tools/mpas_mask_creator.cpp:85:17: error: ‘tuple’ was not declared in this scope
 vector< vector< tuple<string, string, int> > > pointProperties;
                 ^
/usr/projects/climate/mpeterse/repos/tools/pr_test/mesh_tools/mesh_conversion_tools/mpas_mask_creator.cpp:85:42: error: wrong number of template arguments (3, should be at least 1)
 vector< vector< tuple<string, string, int> > > pointProperties;
                                          ^
In file included from /turquoise/usr/projects/hpcsoft/toss3/common/x86_64/gcc/5.3.0/include/c++/5.3.0/vector:64:0,
                 from /usr/projects/climate/mpeterse/repos/tools/pr_test/mesh_tools/mesh_conversion_tools/mpas_mask_creator.cpp:6:
/turquoise/usr/projects/hpcsoft/toss3/common/x86_64/gcc/5.3.0/include/c++/5.3.0/bits/stl_vector.h:214:11: note: provided for ‘template<class _Tp, class _Alloc> class std::vector’
     class vector : protected _Vector_base<_Tp, _Alloc>
           ^
/usr/projects/climate/mpeterse/repos/tools/pr_test/mesh_tools/mesh_conversion_tools/mpas_mask_creator.cpp:85:44: error: template argument 1 is invalid
 vector< vector< tuple<string, string, int> > > pointProperties;
                                            ^
/usr/projects/climate/mpeterse/repos/tools/pr_test/mesh_tools/mesh_conversion_tools/mpas_mask_creator.cpp:85:44: error: template argument 2 is invalid
/usr/projects/climate/mpeterse/repos/tools/pr_test/mesh_tools/mesh_conversion_tools/mpas_mask_creator.cpp:85:46: error: expected unqualified-id before ‘>’ token
 vector< vector< tuple<string, string, int> > > pointProperties;
                                              ^
/usr/projects/climate/mpeterse/repos/tools/pr_test/mesh_tools/mesh_conversion_tools/mpas_mask_creator.cpp:109:9: error: ‘tuple’ was not declared in this scope
 vector< tuple<string, string, int> > extractFeatureProperties( Json::Value properties );
         ^
/usr/projects/climate/mpeterse/repos/tools/pr_test/mesh_tools/mesh_conversion_tools/mpas_mask_creator.cpp:109:34: error: wrong number of template arguments (3, should be at least 1)
 vector< tuple<string, string, int> > extractFeatureProperties( Json::Value properties );
                                  ^
In file included from /turquoise/usr/projects/hpcsoft/toss3/common/x86_64/gcc/5.3.0/include/c++/5.3.0/vector:64:0,
                 from /usr/projects/climate/mpeterse/repos/tools/pr_test/mesh_tools/mesh_conversion_tools/mpas_mask_creator.cpp:6:
/turquoise/usr/projects/hpcsoft/toss3/common/x86_64/gcc/5.3.0/include/c++/5.3.0/bits/stl_vector.h:214:11: note: provided for ‘template<class _Tp, class _Alloc> class std::vector’
     class vector : protected _Vector_base<_Tp, _Alloc>
           ^
/usr/projects/climate/mpeterse/repos/tools/pr_test/mesh_tools/mesh_conversion_tools/mpas_mask_creator.cpp:109:36: error: expected unqualified-id before ‘>’ token
 vector< tuple<string, string, int> > extractFeatureProperties( Json::Value properties );
                                    ^
/usr/projects/climate/mpeterse/repos/tools/pr_test/mesh_tools/mesh_conversion_tools/mpas_mask_creator.cpp:429:9: error: ‘tuple’ was not declared in this scope
 vector< tuple<string, string, int> > extractFeatureProperties( Json::Value properties ){/*{{{*/
         ^
/usr/projects/climate/mpeterse/repos/tools/pr_test/mesh_tools/mesh_conversion_tools/mpas_mask_creator.cpp:429:34: error: wrong number of template arguments (3, should be at least 1)
 vector< tuple<string, string, int> > extractFeatureProperties( Json::Value properties ){/*{{{*/
                                  ^
In file included from /turquoise/usr/projects/hpcsoft/toss3/common/x86_64/gcc/5.3.0/include/c++/5.3.0/vector:64:0,
                 from /usr/projects/climate/mpeterse/repos/tools/pr_test/mesh_tools/mesh_conversion_tools/mpas_mask_creator.cpp:6:
/turquoise/usr/projects/hpcsoft/toss3/common/x86_64/gcc/5.3.0/include/c++/5.3.0/bits/stl_vector.h:214:11: note: provided for ‘template<class _Tp, class _Alloc> class std::vector’
     class vector : protected _Vector_base<_Tp, _Alloc>
           ^
/usr/projects/climate/mpeterse/repos/tools/pr_test/mesh_tools/mesh_conversion_tools/mpas_mask_creator.cpp:429:36: error: expected unqualified-id before ‘>’ token
 vector< tuple<string, string, int> > extractFeatureProperties( Json::Value properties ){/*{{{*/
                                    ^
make[2]: *** [CMakeFiles/MpasMaskCreator.x.dir/mpas_mask_creator.cpp.o] Error 1
make[1]: *** [CMakeFiles/MpasMaskCreator.x.dir/all] Error 2
make: *** [all] Error 2
```
 </p></details>

I'm testing with head right now.

**Comment by mark-petersen** [comment]
This still compiles for both intel and gnu on grizzly, on this PR:
```
cd mesh_tools/mesh_conversion_tools/
make
```
Of course, need to change Makefile for each compiler.

**Comment by xylar** [comment]
Thanks @mark-petersen.  That's quite disappointing.  I'll look into it.

**Comment by xylar** [comment]
@mark-petersen, I can reproduce your error.  If I load the following, I'm able to build fine:
```
$ module list

Currently Loaded Modules:
  1) intel/18.0.5   3) git/2.19.1           5) netcdf-serial/4.4.0
  2) cmake/3.14.0   4) hdf5-serial/1.8.16
```

I really do suspect that `intel/17.0.1` might be too old to support some of the features Doug used in this code.  But that doesn't explain why the code builds fine with the other make file.

**Comment by xylar** [comment]
I see what it is.  In the Makefile, we do explicitly enable the "experimental" C++ features with `-std=c++0x`.  This has become standard in more recent compilers, so the flag isn't needed.  I'll see how to supply this to cmake...

**Comment by xylar** [comment]
@mark-petersen, it should be fixed now.  If you have time, could you test again?

**Pull request merged**